<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Refinehighqualvar by joonan30</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Refinehighqualvar</h1>
      <h2 class="project-tagline">A small tutorial to refine high-quality variant calling</h2>
      <a href="https://github.com/joonan30/RefineHighQualVar" class="btn">View on GitHub</a>
      <a href="https://github.com/joonan30/RefineHighQualVar/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/joonan30/RefineHighQualVar/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h1>
<a id="refine-high-quality-variant-calling" class="anchor" href="#refine-high-quality-variant-calling" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Refine high-quality variant calling</h1>

<h3>
<a id="background" class="anchor" href="#background" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Background</h3>

<p>Despite high-throughput sequencing approaches, variant detection is uncertain in few genomic regions and different sequencing settings - coverage, read length, sequencing depth dosage, and sequencing platform. </p>

<p>Few models have been suggested to amerilorate this problem - i.e. the Variant Quality Score Recalibration (VQSR) by the GATK pipeline. However, such models may not be applied in general sequencing setting, or not fully represent sequencing settings in the different context - targeted sequencing, rare variant discovery, or low-cov sequencing data.</p>

<p>Hence, we utilize emprical truth variant set to examine filtering criteria for high-quality variant calling. </p>

<h3>
<a id="input-data" class="anchor" href="#input-data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Input data</h3>

<p>You will use validated variants as a positive set, and non-validated variants as a negative set. "Validated variants" can be the ones obtained from experimental data, or the ones overlapped among different pipeline/ reference sets. </p>

<p>One example is Zook et al. (2014), who evaluates the criteria of high-quality variant calling. In this study, the authors use the set of concordant variant calls from multiple pipelines and sequencing platforms. Then, using <a href="https://en.wikipedia.org/wiki/Receiver_operating_characteristic">receiver operating characteristic (ROC) curves</a>, they showed how false-positive and true-positive rates change while varying the cutoff for different quality metrics, such as read depth, variant quality score.  </p>

<h3>
<a id="roc-curves" class="anchor" href="#roc-curves" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>ROC curves</h3>

<p>Here we use a simulated dataset to get quality metrics of ~50k variant calls. We set some are validated variants, defined as a positive set for ROC analysis and non-validated ones as a negative set. </p>

<p>When we explore the distribution of each metric (<strong>Figure 1</strong>), some metrics, like GQ and QUAL, have clear distinction between the positive and negative set, so that we can easily infer which threshold will discriminate true-positives (TPs) and true-neagtives (TNs). However, metrics like MQ or DP do not have obvious cutoff for TPs and TNs we may want to use threshold for TPs at the cost of sensitivity.  </p>

<p><img src="https://github.com/joonan30/RefineHighQualVar/blob/master/plot.distribution.png?raw=true" alt="Figure 1"></p>

<p><strong>Figure 1. Distribution of quality metrics between the positive and negative set of variant calls</strong>*</p>

<p>For ROC curve, we use the ROCR package in R, which measures performance and visualize such measurements over a range of cutoffs (<strong>Figure 2</strong>). The script is attached in the repo. </p>

<p><img src="https://github.com/joonan30/RefineHighQualVar/blob/master/plot.roc.png?raw=true" alt="Figure 2"></p>

<p><strong>Figure 2. ROC curves</strong></p>

<h3>
<a id="reference" class="anchor" href="#reference" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Reference</h3>

<p>Zook et al., Nature Biotechnology (2014). Integrating human sequence data sets provides a resource of benchmark SNP and indel genotype calls (<a href="http://www.nature.com/nbt/journal/v32/n3/full/nbt.2835.html">Link</a>)</p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/joonan30/RefineHighQualVar">Refinehighqualvar</a> is maintained by <a href="https://github.com/joonan30">joonan30</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
